lib/encrypt_attributes.rb

```diff
@@ -39,6 +39,21 @@ def encrypted_attribute(name, options={})
         value = send(name)
         value.respond_to?(:empty?) ? !value.empty? : !!value
       end
+
+      # Define methods using serialized accessors option
+      if options[:serialize].is_a?(Hash) && options[:serialize].has_key?(:accessors) && options[:serialize][:accessors].is_a?(Array)
+        options[:serialize][:accessors].map(&:to_sym).each do |accessor|
+          accessor_name = "#{name}_#{accessor}"
+
+          define_method(accessor_name) do
+            send(name)[accessor]
+          end
+
+          define_method("#{accessor_name}=") do |val|
+            send("#{name}=", { "#{accessor}": val })
```

Comment:
This will overwrite anything else in the encrypted hash, so if you have some encrypted data `{ foo: "bar" }` in `post.data`, and you use `post.data_attribute = "baz"`, you will end up with `{ attribute: "baz" }` and loose the original `foo: "bar"` hash/key pair.

I believe this should work:

```ruby
send("#{name}=", { accessor.to_sym => val }.merge!(send(name) || {}))
```